lib/rspec/core/configuration.rb

```diff
@@ -1412,13 +1415,32 @@ def self.delegate_to_ordering_manager(*methods)
       #     end
       #   end
       #
-      #   describe MyClass, :order => :reverse do
+      #   RSpec.describe 'MyClass', :order => :reverse do
       #     # ...
       #   end
       #
       # @note Pass the symbol `:global` to set the ordering strategy that
       #   will be used to order the top-level example groups and any example
       #   groups that do not have declared `:order` metadata.
+      #
+      # @example
+      #   RSpec.configure do |rspec|
+      #     rspec.register_ordering :global do |examples|
+      #       acceptance, other = examples.partition do |example|
+      #         example.metadata[:type] == :acceptance
```

Comment:
> If complexity is problem, the API has to change.

I don't agree.  RSpec has never had the goal of providing APIs to make anything a user might want to do easy.  That's fool's errand and we have neither the time nor desire to pursue that.

We do want to provide APIs that are sufficiently flexible that anything a user might want to do is _possible_ without brittle monkey patches.  That's what the ordering API is about: it provides the flexibility for users to order things with any arbitrary logic.  How simple or complex the logic is is up to the user.  You as a user can define the logic in a custom class using a `<=>` operator, and write your block to delegate to that if you prefer.  I also don't think it makes sense to provide a CLI option to select a custom ordering strategy.  The options supported from the CLI are intentionally limited.  We don't want `rspec --help` to print off a long wall of text that makes it intimidating or hard for a user to find the options they care about.